### PR TITLE
Write some basic documentation for the various iroh crates

### PR DESCRIPTION
**Pull request by faassen**
When I was going through the iroh crates I noticed that the readmes were rather minimal or absent. Now each iroh crate (except `iroh` itself which is still empty) contains a basic README with some explanation of what's going on on. This is to the best of my knowledge, so this is also a way to test my knowledge - please comment where I'm wrong to improve my understanding.

### CODE REVIEW

File: iroh-car/README.md [hunk 1]
```diff
@@ -1,9 +1,15 @@
 # iroh-car
 
-> [CAR file](https://ipld.io/specs/transport/car/) support for iroh.
+[CAR file](https://ipld.io/specs/transport/car/) support for iroh. "CAR" stands
+for Content Addressable aRchives. A CAR file typically contains a serialized
+representation of an [IPLD
+DAG](https://docs.ipfs.tech/concepts/merkle-dag/#merkle-directed-acyclic-graphs-dags),
+though is general enough to citnain arbitrary IPLD blocks.
```
Comment (Winterhuman) [inline iroh-car/README.md:7]: "cintain" -> "contain"

Comment (faassen) [reply iroh-car/README.md:7]: Sorry, pressed the wrong button to rerequest review, will fix soon.

Comment (faassen) [reply iroh-car/README.md]: Fixed for real now!

File: iroh-gateway/src/headers.rs [hunk 2]
```diff
@@ -61,7 +61,6 @@ pub fn add_cache_control_headers(headers: &mut HeaderMap, metadata: Metadata) {
             HeaderValue::from_str(&lmdt.to_string()).unwrap(),
         );
     } else {
-        headers.insert(LAST_MODIFIED, HeaderValue::from_str("0").unwrap());
```
Comment (dignifiedquire) [inline iroh-gateway/src/headers.rs:64]: @Arqu @faassen what's the deal here?

File: iroh-util/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# iroh-util
+
+Utility functions for [iroh](https://github.com/n0-computer/iroh). This
+provides shared functionality to be used in other iroh crates.
+
+## How to run
+
+```sh
+# From the root of the workspace
+> cargo run --release -p iroh-store
```
Comment (dignifiedquire) [inline iroh-util/README.md:10]: copy paste error

File: iroh-gateway/src/headers.rs (hunk 2 above)
Comment (faassen) [reply iroh-gateway/src/headers.rs:64]: I think I might've squashed one too many a commit or something.

Comment (faassen) [reply iroh-gateway/src/headers.rs:64]: Ah, it *doesn't* show up in the diff, only in the commits. Must be me bungling fetching upstream or something.

**Comment by faassen** [comment]
I'm closing this PR in favor of https://github.com/n0-computer/iroh/pull/241 as I managed to get things really tangled up trying to rebase inexpertly.

**Pull request closed without merging**